### PR TITLE
Add complete task-8

### PR DESCRIPTION
**Pull request by bugagashinka**
_Complete the form below and remove this line._

### js-oop-task

[Demo](https://bugagashinka.github.io/frontend-nanodegree-arcade-game/) |
[Code base](https://github.com/bugagashinka/frontend-nanodegree-arcade-game)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/bugagashinka/js-oop-frogger/js/app.js [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+  this.lifeTime = generateNum(2, 4);
+  this.sprite = null;
+};
+
+function showProfit(imgs) {
+  if (this.show) return;
+  this.show = true;
+
+  var imgScope = imgs.length - 1;
+  if (player.lifes < PLAYER_MAX_LIFE) {
```
Comment (OleksiyRudenko) [inline submissions/bugagashinka/js-oop-frogger/js/app.js]: It is not good for a class to be aware of a particular instance.
It is against design patterns to refer to global variables. The only exclusions are constants and library functions.
Please, fix. Here and across the code.

File: submissions/bugagashinka/js-oop-frogger/js/app.js [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+    });
+  } else if (direction == 'right') {
+    this.move({
+      x: this.x + 100,
+      y: this.y,
+    });
+  } else if (direction == 'down') {
+    this.move({
+      x: this.x,
+      y: this.y + 80,
```
Comment (OleksiyRudenko) [inline submissions/bugagashinka/js-oop-frogger/js/app.js]: Please, declare "magic" numbers here and elsewhere across the code as constants.

File: submissions/bugagashinka/js-oop-frogger/js/app.js [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+var allEnemies = [
+    new Enemy({ x: 0, y: 55 }, 80),
+    new Enemy({ x: 0, y: 145 }, 100),
+    new Enemy({ x: -250, y: 55 }, 80),
+    new Enemy({ x: -250, y: 230 }, 120),
+  ],
+  // Place the player object in a variable called player
+  player = new Player(playerStartPos),
+  profit = new Profit();
+
```
Comment (OleksiyRudenko) [inline submissions/bugagashinka/js-oop-frogger/js/app.js]: Cannot easily identify a method in charge of player-enemies collision handling.
Please, make this part of game strategy clear and identifiable.

File: submissions/bugagashinka/js-oop-frogger/js/app.js [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+  this.isGrabbed = false;
+  this.show = false;
+  this.x = -50;
+  this.y = 0;
+  this.lifeTime = generateNum(2, 4);
+  this.sprite = null;
+};
+
+function showProfit(imgs) {
+  if (this.show) return;
```
[the remark this answers is not in the document]
Comment (bugagashinka) [reply submissions/bugagashinka/js-oop-frogger/js/app.js]: Thanks for review and advice, I tried make these improvements.

File: submissions/bugagashinka/js-oop-frogger/js/app.js (hunk 1 above)
Comment (bugagashinka) [reply submissions/bugagashinka/js-oop-frogger/js/app.js]: Done

File: submissions/bugagashinka/js-oop-frogger/js/app.js (hunk 2 above)
Comment (bugagashinka) [reply submissions/bugagashinka/js-oop-frogger/js/app.js]: Done

File: submissions/bugagashinka/js-oop-frogger/js/app.js (hunk 3 above)
Comment (bugagashinka) [reply submissions/bugagashinka/js-oop-frogger/js/app.js]: Done

**Comment by zonzujiro** [comment]
@bugagashinka pls, resolve outdated discussions

**Pull request merged**